contrib/clojure-package/examples/imclassification/src/imclassification/train_mnist.clj

```diff
@@ -96,18 +75,33 @@
     (do
       (println "Starting Training of MNIST ....")
       (println "Running with context devices of" devs)
-      (let [_mod (m/module (get-symbol) {:contexts devs})]
-        (m/fit _mod {:train-data train-data
-                    :eval-data test-data
+      (resource-scope/with-let [_mod (m/module (get-symbol) {:contexts devs})]
+        (-> _mod
+            (m/fit {:train-data (mx-io/mnist-iter {:image (str data-dir "train-images-idx3-ubyte")
```

Comment:
It will work if they are `defs` outside, but you will still get the undisposed ndarray warning. I refactored to move it out to functions which does work and looks better too :)